auth/src/main/java/com/github/twitch4j/auth/providers/TwitchIdentityProvider.java

```diff
@@ -38,6 +38,41 @@ public TwitchIdentityProvider(String clientId, String clientSecret, String redir
         this.scopeSeperator = "+"; // Prevents a URISyntaxException when creating a URI from the authUrl
     }
 
+    /**
+     * Checks whether an {@link OAuth2Credential} is valid.
+     * <p>
+     * Expired tokens will yield false (assuming the network request succeeds).
+     *
+     * @param credential the OAuth credential to check
+     * @return whether the token is valid, or empty if the network request did not succeed
+     */
+    public Optional<Boolean> isCredentialValid(OAuth2Credential credential) {
+        if (credential == null || credential.getAccessToken() == null || credential.getAccessToken().isEmpty())
+            return Optional.of(false);
+
+        try {
+            // build request
+            Request request = new Request.Builder()
+                .url("https://id.twitch.tv/oauth2/validate")
+                .header("Authorization", "OAuth " + credential.getAccessToken())
+                .build();
+
+            // perform call
+            Response response = HTTP_CLIENT.newCall(request).execute();
```

Comment:
These requests will possibly not use the auto configured proxy as other identity provider requests. The same applies to other requests in this class.

I added a httpClient in the OAuth2IdentityProvider which we can use.